pyproject.toml

```diff
@@ -7,7 +7,7 @@ name = "hayhooks"
 dynamic = ["version"]
 description = 'Grab and deploy Haystack pipelines'
 readme = "README.md"
-requires-python = ">=3.7,<3.13"
+requires-python = ">=3.9,<3.13"
```

Comment:
Since we are touching pyproject.toml, let's introduce `uv` as the installer. It should make installation and tests faster.

See https://github.com/deepset-ai/haystack/blob/c306bee66563c23a08769c7bd6a56f01d5cac138/pyproject.toml#L63C1-L64C17